### PR TITLE
feat: add `validation-message` and `validation-icon` attributes to form components for new internal `calcite-input-message`

### PR DESCRIPTION
**Pull request by benelan**
**Related Issue:** #8057

## Summary

Add an internal `calcite-input-message` to the following components 
- combobox
- input-date-picker
- input-number
- input-text
- input-time-picker
- input-time-zone
- input
- select
- text-area

The new properties will be used to replace the browser's popover element that displays validation messages on form submission (#8000).

### CODE REVIEW

**Comment by anveshmekala** [comment]
Is it possible to avoid using `message` prefix for props since we have `messages` and `messageOverrides` props as part of translation effort. Thoughts?

**Comment by benelan** [comment]
> Is it possible to avoid using `message` prefix for props since we have `messages` and `messageOverrides` props as part of translation effort. Thoughts?

Any other ideas? Maybe `validationText` (or `validationMessage` if suffix is okay) and `validationIcon`? cc @jcfranco

**Comment by ashetland** [comment]
> > Is it possible to avoid using `message` prefix for props since we have `messages` and `messageOverrides` props as part of translation effort. Thoughts?
> 
> Any other ideas? Maybe `validationText` (or `validationMessage` if suffix is okay) and `validationIcon`? cc @jcfranco

`statusMessage`?

**Comment by jcfranco** [comment]
> Any other ideas? Maybe validationText (or validationMessage if suffix is okay) and validationIcon? cc @jcfranco

If the message and icon are only displayed when the component goes through validation, I think `validationMessage` and `validationIcon` fit best. IMO, their intent is clearer vs using `statusMessage`/`statusIcon`. It would also align nicely w/ ElementInternal's [`validationMessage` prop](https://developer.mozilla.org/en-US/docs/Web/API/ElementInternals/validationMessage).

File: packages/calcite-components/src/components/select/select.e2e.ts [hunk 1]
```diff
@@ -22,7 +22,7 @@ describe("calcite-select", () => {
   `;
 
   describe("renders", () => {
-    renders(simpleTestMarkup, { display: "flex" });
+    renders(simpleTestMarkup, { display: "inline" });
```
Comment (jcfranco) [inline packages/calcite-components/src/components/select/select.e2e.ts:25]: The host display is `flex`, so this test should be failing. Can you double check?

**Review by jcfranco — APPROVED** [review]
✔️✔️✔️✔️✔️✔️✔️✔️✔️✔️✔️✔️✔️✔️✔️✔️✔️✔️✔️✔️✔️✔️✔️✔️✔️✔️✔️✔️
✔️✅✔️✔️✔️✅✔️✔️✅✅✔️✔️✅✅✅✔️✔️✅✅✅✔️✅✅✅✅✔️✅✔️
✔️✅✅✔️✔️✅✔️✅✔️✔️✅✔️✔️✅✔️✔️✅✔️✔️✔️✔️✅✔️✔️✔️✔️✅✔️
✔️✅✔️✅✔️✅✔️✅✔️✔️✅✔️✔️✅✔️✔️✅✔️✔️✔️✔️✅✅✅✔️✔️✅✔️
✔️✅✔️✔️✅✅✔️✅✔️✔️✅✔️✔️✅✔️✔️✅✔️✔️✔️✔️✅✔️✔️✔️✔️✔️✔️
✔️✅✔️✔️✔️✅✔️✔️✅✅✔️✔️✅✅✅✔️✔️✅✅✅✔️✅✅✅✅✔️✅✔️
✔️✔️✔️✔️✔️✔️✔️✔️✔️✔️✔️✔️✔️✔️✔️✔️✔️✔️✔️✔️✔️✔️✔️✔️✔️✔️✔️✔️

As a follow-up, can we extract and reuse the common rendering logic and possibly helper CSS class?

**Pull request merged**